### PR TITLE
Fix defaultQueryTimeout

### PR DESCRIPTION
**Pull request by awelsh93**
Fix for #5731 

Made a change so that SetAndVerifyContextQueryRunner will set query fail time after the default query timeout is set in the query context.

### CODE REVIEW

**Comment by gianm** [comment]
Hi @awelsh93, could you please fill out the project CLA so we can accept this contribution? It's at: http://druid.io/community/cla.html

File: server/src/main/java/io/druid/client/DirectDruidClient.java [hunk 1]
```diff
@@ -116,28 +116,14 @@
...
   }
 
-  public static Map<String, Object> makeResponseContextForQuery(Query query, long startTimeMillis)
-  {
-    final Map<String, Object> responseContext = new ConcurrentHashMap<>();
-    responseContext.put(
-        DirectDruidClient.QUERY_FAIL_TIME,
-        startTimeMillis + QueryContexts.getTimeout(query)
-    );
-    responseContext.put(
```
Comment (gianm) [inline server/src/main/java/io/druid/client/DirectDruidClient.java]: What is going to add `QUERY_TOTAL_BYTES_GATHERED` now?

File: server/src/main/java/io/druid/server/SetAndVerifyContextQueryRunner.java [hunk 2]
```diff
@@ -35,11 +37,13 @@
...
   private final ServerConfig serverConfig;
   private final QueryRunner<T> baseRunner;
+  private final long startTimeMillis;
 
-  public SetAndVerifyContextQueryRunner(ServerConfig serverConfig, QueryRunner<T> baseRunner)
+  public SetAndVerifyContextQueryRunner(ServerConfig serverConfig, QueryRunner<T> baseRunner, long startTimeMillis)
   {
     this.serverConfig = serverConfig;
     this.baseRunner = baseRunner;
+    this.startTimeMillis = startTimeMillis;
```
Comment (gianm) [inline server/src/main/java/io/druid/server/SetAndVerifyContextQueryRunner.java]: This is always set to `System.currentTimeMillis()`, might as well set it in the constructor so callers don't have to pass that in. Unless you are intending to use the flexibility to write some unit tests -- in which case go for it.

File: server/src/main/java/io/druid/server/SetAndVerifyContextQueryRunner.java [hunk 3]
```diff
@@ -53,7 +57,7 @@ public SetAndVerifyContextQueryRunner(ServerConfig serverConfig, QueryRunner<T>
 
   public Query<T> withTimeoutAndMaxScatterGatherBytes(Query<T> query, ServerConfig serverConfig)
   {
-    return QueryContexts.verifyMaxQueryTimeout(
+    Query<T> new_query = QueryContexts.verifyMaxQueryTimeout(
```
Comment (gianm) [inline server/src/main/java/io/druid/server/SetAndVerifyContextQueryRunner.java]: `newQuery` is more Druid-like spelling.

**Comment by awelsh93** [comment]
Hi @gianm, thank you for taking at look at this and leaving comments. I've addressed these comments and I have signed the CLA.

File: server/src/main/java/io/druid/client/DirectDruidClient.java [hunk 4]
```diff
@@ -116,21 +116,17 @@
   private final boolean isSmile;
 
   /**
-   * Removes the magical fields added by {@link #makeResponseContextForQuery(Query, long)}.
+   * Removes the magical fields added by {@link #makeResponseContextForQuery()}.
    */
   public static void removeMagicResponseContextFields(Map<String, Object> responseContext)
   {
     responseContext.remove(DirectDruidClient.QUERY_FAIL_TIME);
```
Comment (jihoonson) [inline server/src/main/java/io/druid/client/DirectDruidClient.java]: This line should also be removed.

**Comment by jihoonson** [comment]
It would be great if we can add a unit test for this bug, but not sure how to do it yet.

**Pull request merged**